drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/src/main/java/org/drools/workbench/screens/scenariosimulation/kogito/client/editor/strategies/KogitoDMODataManagementStrategy.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.scenariosimulation.kogito.client.editor.strategies;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.SortedMap;
+
+import org.drools.workbench.screens.scenariosimulation.client.commands.ScenarioSimulationContext;
+import org.drools.workbench.screens.scenariosimulation.client.editor.strategies.AbstractDMODataManagementStrategy;
+import org.drools.workbench.screens.scenariosimulation.client.enums.GridWidget;
+import org.drools.workbench.screens.scenariosimulation.client.rightpanel.TestToolsView;
+import org.drools.workbench.screens.scenariosimulation.kogito.client.fakes.KogitoAsyncPackageDataModelOracle;
+import org.drools.workbench.screens.scenariosimulation.model.ScenarioSimulationModelContent;
+import org.drools.workbench.screens.scenariosimulation.model.typedescriptor.FactModelTree;
+import org.kie.soup.project.datamodel.oracle.ModelField;
+import org.uberfire.backend.vfs.ObservablePath;
+
+public class KogitoDMODataManagementStrategy extends AbstractDMODataManagementStrategy {
+
+    protected KogitoAsyncPackageDataModelOracle kogitoOracle;
+
+    public KogitoDMODataManagementStrategy(KogitoAsyncPackageDataModelOracle kogitoOracle) {
+        this.kogitoOracle = kogitoOracle;
+    }
+
+    @Override
+    public void manageScenarioSimulationModelContent(ObservablePath currentPath, ScenarioSimulationModelContent toManage) {
+        model = toManage.getModel();
+        kogitoOracle.init(currentPath);
+    }
+
+    @Override
+    public boolean isADataType(String value) {
+        return kogitoOracle != null && Arrays.asList(kogitoOracle.getFactTypes()).contains(value);
```

Comment:
I haven't found an implementation of `getFactTypes()`. Do we have one? Further more should we check its not nul before we pass it to `Arrays.asList`?

Reply:
@jomarko At the moment we don't have an implementation of KogitoOracle. This because in business central this part is all on backend side, and it's easy task write a KogitoOracle for a client side. For the near future it will remain not-implemented.

Reply:
@yesamer in other words, won't `Arrays.asList(kogitoOracle.getFactTypes())` throw an exception when `kogitoOracle.getFactTypes()` returns null?

Reply:
@jomarko Ok,  I've got your point, and you're right, a NPE will be launched in that case. I'll check to understand hot to manage it.

Reply:
ok, I guess this is last point, let me know when you are done with investigation.

Reply:
@jomarko Found it! The implementation of this interface it will present on `-testing` and `-runtime` module. THerefore, it's fine for now!